presto-main/src/main/java/com/facebook/presto/twitter/logging/QueryScriber.java

```diff
@@ -0,0 +1,169 @@
+import com.facebook.presto.event.query.QueryEventHandler;
+import com.google.common.base.Optional;
+import com.twitter.logging.BareFormatter$;
+import com.twitter.logging.Level;
+import com.twitter.logging.QueueingHandler;
+import com.twitter.logging.ScribeHandler;
+import com.twitter.presto.thriftjava.QueryState;
+import io.airlift.log.Logger;
+import org.apache.thrift.TBase;
+import org.apache.thrift.TException;
+import org.apache.thrift.TSerializer;
+
+import java.util.Base64;
+import java.util.logging.LogRecord;
+
+/**
+ * Class that scribes query completion events
+ */
+public class QueryScriber implements QueryEventHandler<QueryCompletionEvent>
+{
+    private static final String SCRIBE_CATEGORY = "test_presto_query_complete";
+    private static final int MAX_QUEUE_SIZE = 1000;
+
+    private static final Logger log = Logger.get(QueryScriber.class);
+
+    private QueueingHandler queueingHandler;
+
+    // TSerializer is not thread safe
+    private final ThreadLocal<TSerializer> serializer = new ThreadLocal<TSerializer>()
+    {
+        @Override protected TSerializer initialValue()
+        {
+          return new TSerializer();
+        }
+    };
+
+    public QueryScriber()
+    {
+        ScribeHandler scribeHandler = new ScribeHandler(
+                ScribeHandler.DefaultHostname(),
+                ScribeHandler.DefaultPort(),
+                SCRIBE_CATEGORY,
+                ScribeHandler.DefaultBufferTime(),
+                ScribeHandler.DefaultConnectBackoff(),
+                ScribeHandler.DefaultMaxMessagesPerTransaction(),
+                ScribeHandler.DefaultMaxMessagesToBuffer(),
+                BareFormatter$.MODULE$,
+                scala.Option.apply((Level) null));
+        queueingHandler = new QueueingHandler(scribeHandler, MAX_QUEUE_SIZE);
+    }
+
+    @Override
+    public void handle(QueryCompletionEvent event)
+    {
+        com.twitter.presto.thriftjava.QueryCompletionEvent thriftEvent = toThriftQueryCompletionEvent(event);
+        Optional<String> message = serializeThriftToString(thriftEvent);
+
+        if (message.isPresent()) {
+          LogRecord logRecord = new LogRecord(Level.ALL, message.get());
+          queueingHandler.publish(logRecord);
+        }
+        else {
+          log.warn("Unable to serialize QueryCompletionEvent: " + event);
+        }
+    }
+
+    /**
+    * Serialize a thrift object to bytes, compress, then encode as a base64 string.
+    */
+    private Optional<String> serializeThriftToString(TBase thriftMessage)
+    {
+        try {
+          return Optional.of(
+                  Base64.getEncoder().encodeToString(serializer.get().serialize(thriftMessage)));
+        }
+        catch (TException e) {
+            log.warn(e, "Could not serialize thrift object" + thriftMessage);
+          return Optional.absent();
+        }
+    }
+
+    private static com.twitter.presto.thriftjava.QueryCompletionEvent toThriftQueryCompletionEvent(QueryCompletionEvent event)
```

Comment:
No, because the issue is that we have two classes with the same name (which I think is ok in this case):
```
com.facebook.presto.event.query.QueryCompletionEvent
com.twitter.presto.thriftjava.QueryCompletionEvent
```
Java doesn't provide a mechanism to alias packages like scala. :(